File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int i = 0;
        int j = 0;
        int max = 0;
        List<Character> arr = new ArrayList<>(); 
        int n = s.length();
        while(j < n){
            if(!arr.contains(s.charAt(j))){
                arr.add(s.charAt(j));
                j++;
                max = Math.max(max, arr.size());
            }else{
                arr.remove(Character.valueOf(s.charAt(i)));
                i++;
            }
        }
        return max;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        n = len(s)
        a = set({})
        i, j = 0, 1
        if n <= 1:
            return n
        a.add(s[0])
        x = 1
        while j < n:
            while s[j] in a:
                a.discard(s[i])
                i += 1
            a.add(s[j])
            j += 1
            x = max(x,(j-i))
        return x

        